iroh-net/src/magicsock/node_map.rs

```diff
@@ -536,6 +571,29 @@ impl NodeMapInner {
     }
 }
 
+/// Stream returning `ConnectionTypes`
+#[derive(Debug)]
+pub struct ConnectionTypeStream {
+    initial: Option<ConnectionType>,
+    inner: watchable::WatcherStream<ConnectionType>,
+}
+
+impl Stream for ConnectionTypeStream {
+    type Item = ConnectionType;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this = &mut *self;
+        if let Some(initial_conn_type) = this.initial.take() {
+            return Poll::Ready(Some(initial_conn_type));
+        }
+        match Pin::new(&mut this.inner).poll_next(cx) {
+            Poll::Pending => Poll::Pending,
+            Poll::Ready(Some(conn_type)) => Poll::Ready(Some(conn_type)),
+            Poll::Ready(None) => Poll::Ready(None),
+        }
```

Comment:
Isn't this the same as directly `Pin::new(&mut this.inner).poll_next(cx)`?

Reply:
lmao yep thank you.